### ISSUE TITLE
Submariner gateway crashed: Libreswan unrecognized option '--ikeport' 

### ISSUE THREAD

**Issue by manosnoam** [submariner-io/submariner#776]
On Submariner 0.6.0 with non-globalnet + libreswan I saw that HA on Cluster A (AWS) was left on passive status.

Looking at the pods on this cluster, I saw that Gateway pod in CrashLoopBackOff, and was restarted 5 times during ~5 minutes.
The culprit seems to be in Libreswan:

/usr/libexec/ipsec/pluto: unrecognized option '--ikeport'
For usage information: /usr/libexec/ipsec/pluto --help
Libreswan v3.30-1472-g4a4925333f-main
F0826 22:38:16.405002       1 libreswan.go:365] Pluto exited: exit status 1

Full test scenario:
https://qe-jenkins-csb-skynet.cloud.paas.psi.redhat.com/job/debug_job/296/Test-Report/

Gateway pod log:
[libreswan_no_ikeport.log](https://github.com/submariner-io/submariner/files/5136581/libreswan_no_ikeport.log)

**What happened**:
```
$ subctl join --clusterid nmanos-cluster-a   ./broker-info.subm --cable-driver libreswan   --ikeport 501 --nattport 4501 --version libreswan-git --enable-pod-debugging

* ./broker-info.subm says broker is at: https://api.nmanos-cluster-a.devcluster.openshift.com:6443
* There are 1 labeled nodes in the cluster:
  - ip-10-0-74-54.us-east-2.compute.internal
 • Discovering network details  ...
    Discovered network details:
        Network plugin:  OpenShift
        Service CIDRs:   [172.30.0.0/16]
        Cluster CIDRs:   [10.128.0.0/14]
 ✓ Discovering network details
 • Discovering multi cluster details  ...
 • Validating Globalnet configurations  ...
 ✓ Validating Globalnet configurations
 ✓ Discovering multi cluster details
 • Deploying the Submariner operator  ...
 ✓ Deploying the Submariner operator
 ✓ Created operator CRDs
 ✓ Created operator namespace: submariner-operator
 ✓ Created operator service account and role
 ✓ Updated the privileged SCC
 ✓ Created lighthouse CRDs
 ✓ Created lighthouse service account and role
 ✓ Updated the privileged SCC
 ✓ Created Lighthouse service accounts and roles
 ✓ Deployed the operator successfully
 • Creating SA for cluster  ...
 ✓ Creating SA for cluster
 • Deploying Submariner  ...
 ✓ Deploying Submariner
 ✓ Submariner is up and running


$ oc describe Gateway -n submariner-operator

Name:         ip-10-0-74-54
Namespace:    submariner-operator
Labels:       <none>
Annotations:  update-timestamp: 1598481496
API Version:  submariner.io/v1
Kind:         Gateway
Metadata:
  Creation Timestamp:  2020-08-26T22:35:15Z
  Generation:          1
  Managed Fields:
    API Version:  submariner.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:update-timestamp:
      f:status:
        .:
        f:connections:
        f:haStatus:
        f:localEndpoint:
          .:
          f:backend:
          f:cable_name:
          f:cluster_id:
          f:hostname:
          f:nat_enabled:
          f:private_ip:
          f:public_ip:
          f:subnets:
        f:statusFailure:
        f:version:
    Manager:         submariner-engine
    Operation:       Update
    Time:            2020-08-26T22:38:16Z
  Resource Version:  28849
  Self Link:         /apis/submariner.io/v1/namespaces/submariner-operator/gateways/ip-10-0-74-54
  UID:               af4f3164-5274-42b0-be3f-cbb2796505d6
Status:
  Connections:
  Ha Status:  passive
  Local Endpoint:
    Backend:      libreswan
    cable_name:   submariner-cable-nmanos-cluster-a-10-0-74-54
    cluster_id:   nmanos-cluster-a
    Hostname:     ip-10-0-74-54
    nat_enabled:  true
    private_ip:   10.0.74.54
    public_ip:    18.221.13.96
    Subnets:
      172.30.0.0/16
      10.128.0.0/14
  Status Failure:  
  Version:         v0.5.0-9-ga1627f5
Events:            <none>

$ oc get pods -n submariner-operator --show-labels

NAME                                             READY   STATUS             RESTARTS   AGE     LABELS
submariner-gateway-dqtv7                         0/1     CrashLoopBackOff   5          4m33s   app=submariner-engine,controller-revision-hash=66b97c6757,pod-template-generation=1
submariner-lighthouse-agent-6b7758664b-jvql8     1/1     Running            0          4m31s   app=submariner-lighthouse-agent,component=submariner-lighthouse,pod-template-hash=6b7758664b
submariner-lighthouse-coredns-76fcbb6dc5-bg59b   1/1     Running            0          4m31s   app=submariner-lighthouse-coredns,component=submariner-lighthouse,pod-template-hash=76fcbb6dc5
submariner-lighthouse-coredns-76fcbb6dc5-bnlc4   1/1     Running            0          4m31s   app=submariner-lighthouse-coredns,component=submariner-lighthouse,pod-template-hash=76fcbb6dc5
submariner-operator-74886c944c-hdnq9             1/1     Running            0          4m54s   name=submariner-operator,pod-template-hash=74886c944c
submariner-routeagent-285h6                      1/1     Running            0          4m32s   app=submariner-routeagent,component=routeagent,controller-revision-hash=889754797,pod-template-generation=1
submariner-routeagent-7tmk6                      1/1     Running            0          4m32s   app=submariner-routeagent,component=routeagent,controller-revision-hash=889754797,pod-template-generation=1
submariner-routeagent-lfsl4                      1/1     Running            0          4m32s   app=submariner-routeagent,component=routeagent,controller-revision-hash=889754797,pod-template-generation=1
submariner-routeagent-qdd56                      1/1     Running            0          4m32s   app=submariner-routeagent,component=routeagent,controller-revision-hash=889754797,pod-template-generation=1

$ oc get events -A
...
ubmariner-operator                                <unknown>   Normal    Scheduled                                    pod/submariner-gateway-dqtv7                                              Successfully assigned submariner-operator/submariner-gateway-dqtv7 to ip-10-0-74-54.us-east-2.compute.internal
submariner-operator                                4m36s       Normal    Pulling                                      pod/submariner-gateway-dqtv7                                              Pulling image "quay.io/submariner/submariner:libreswan-git"
submariner-operator                                4m18s       Normal    Pulled                                       pod/submariner-gateway-dqtv7                                              Successfully pulled image "quay.io/submariner/submariner:libreswan-git"
submariner-operator                                2m48s       Normal    Created                                      pod/submariner-gateway-dqtv7                                              Created container submariner
submariner-operator                                2m48s       Normal    Started                                      pod/submariner-gateway-dqtv7                                              Started container submariner
submariner-operator                                2m48s       Normal    Pulled                                       pod/submariner-gateway-dqtv7                                              Container image "quay.io/submariner/submariner:libreswan-git" already present on machine
submariner-operator                                2m18s       Warning   BackOff                                      pod/submariner-gateway-dqtv7                                              Back-off restarting failed container
submariner-operator                                4m37s       Normal    SuccessfulCreate                             daemonset/submariner-gateway                                              Created pod: submariner-gateway-dqtv7


```

**Environment**:
subctl version: v0.6.0-rc0-14-g531b845
AWS cluster A (OCP Version 4.5.7)
OSP cluster B (OCP Version 4.4.3)

**Comment by skitt** [comment]
The `libreswan-git` Submariner engine image was generated without the Libreswan 4 patch; I’ve updated the images, please try this again.

**Comment by manosnoam** [comment]
Thanks @skitt ,
this is now resolved with subctl version: v0.6.0-rc0-15-g898b23d
Closing bug.
